lib/nethash.cpp

```diff
@@ -1,13 +1,30 @@
-#include "backends/p4tools/modules/testgen/targets/bmv2/contrib/bmv2_hash/calculations.h"
+#include "lib/nethash.h"
 
 #include <arpa/inet.h>
 
 #include <algorithm>
 
-namespace P4Tools::P4Testgen::Bmv2 {
+namespace nethash {
 
-/* generating from my Python script gen_crc_tables inspired from the C code at:
-   http://www.barrgroup.com/Embedded-Systems/How-To/CRC-Calculation-C-Code */
+/* This code was adapted from:
```

Comment:
Make this comment part of the header?

Reply:
Actually I think that both this, and the license snipped with (c) Barefoot should be in the .cpp. The reason is that is the code that has that source. The declarations are kind of "on top of that". Or we can have the disclaimer at both, but I think it should be at the implementation and so should be the license/copyright.